### ISSUE TITLE
Extra settings for the Prometheus datasource when used for Thanos or other LTS solutions

### ISSUE THREAD

**Issue by bergquist** [grafana/grafana#16665]
Thanos offers extra settings that can be very useful when looking at long term data such as `downsampling`, `timeout`, `partial_response`, `dedup`

Instead of building a new datasource plugin for Thanos which would be a lot of code duplication, we should add a setting to the Prometheus datasource where LTS storage alternatives can be configured. Query editors and other parts of Grafana can then show those when it makes sense instead of duplicating all other aspects of the Prometheus datasource.

**Comment by povilasv** [comment]
I'll try to implement something to what we discussed in https://github.com/grafana/grafana/pull/16574 .
So you can assign this to me. Also I'll let thanos community know aboiut this.

Could you help me on the UX bits?

```
  jsonData:
    # enable Thanos
    long_term_storage: thanos
    deduplication: true
    timeout: 2m
    downsampling: false
    partial_response: false
```

Maybe we should prefix flags? instead of `deduplication` `thanos_deduplication: true` or do nested structs?

**Comment by tomwilkie** [comment]
SGTM; it would be better long term to allow code reuse between datasources, and have a separate datasource for Thanos (or Cortex).  But for now, adding it to the Prometheus datasource seems fine.

**Comment by davkal** [comment]
UX: I'd add a Long Term Storage section similar to Elastic's custom area and then have a dropdown that defaults to None. When Thanos is selected from the dropdown, additional fields pop up.
Try adding an Elastic datasource in Grafana to see what I mean (you don't need to actually have one). 
Here is the config page code: https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/elasticsearch/partials/config.html and then see the config_ctrl.js one level up on how to read/set those values.

**Comment by povilasv** [comment]
I'm really horrible at frontend stuff :D  so right now I'll do a PR with config thingy.  If it isn't too complicated will try to do js / html

**Comment by davkal** [comment]
Fixed via #19121
